### PR TITLE
Test Checklist: Adding more consistency

### PR DESCRIPTION
**Pull request by pachlava**
During work on test checklist [improvement](https://github.com/Automattic/simplenote-electron/pull/2852) for Electron, a few missing tests were found for macOS, which are present in other platforms. This PR adds them.

### Fix
Added several tests that were absent in macOS.

### Test
New extended test checklist looks lit 🔥 

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
These changed do not require release notes

### CODE REVIEW

**Review by eshurakov — APPROVED** [review]
Thank you! Looks great 👍 

:shipit:

**Comment by pachlava** [comment]
@eshurakov thank you! I wonder why `Verify App Store Target Builds` job is not executed - should anything be done with it?

**Comment by eshurakov** [comment]
> I wonder why Verify App Store Target Builds job is not executed - should anything be done with it?

@pachlava Unfortunately I don't know. Could you help us @mokagio?

File: TESTING-CHECKLIST.md [hunk 1]
```diff
@@ -20,9 +20,9 @@
 - [ ] Markdown setting syncs to/from other device
 - [ ] Preview mode disappears/reappears when receiving remote changes to markdown setting
 - [ ] Note pinning syncs immediately to/from other device
-- [ ] Note pinning works regardless if clicking in list view of from note info
-- [ ] Viewing history on one deivce leaves note unchanged on other device
-- [ ] Restoring history immediately syncs note from both directions
+- [ ] Note pinning works regardless if selecting in list view or from note info
```
Comment (mokagio) [inline TESTING-CHECKLIST.md]:
```suggestion
- [ ] Note pinning works regardless if done from list view or note info
```

File: TESTING-CHECKLIST.md [hunk 2]
```diff
@@ -65,12 +65,12 @@
 
 - [ ] Can toggle sidebar
 - [ ] Can change analytics sharing setting
-- [ ] With wide editor screen, toggling `View` > `Line Length` between `Narrow` and `Full` removes and adds border around note content appropriately and immediately.
 - [ ] Changing `Note Display` mode immediately updates and reflects in note list
 - [ ] Changing `Note Sorting` immediately updates and reflects in note list for each sort type
 - [ ] Changing `Tag Sorting` immediately updates and reflects in tag list
 - [ ] For each sort type the pinned notes appear first in the note list
 - [ ] Changing `Theme` immediately updates app for desired color scheme
+- [ ] With wide editor screen, toggling `View` > `Line Length` between `Narrow` and `Full` removes and adds border around note content appropriately and immediately.
```
Comment (mokagio) [inline TESTING-CHECKLIST.md]: Nipick _None of the other items end with a `.` 😅 🔍 

```suggestion
- [ ] With wide editor screen, toggling `View` > `Line Length` between `Narrow` and `Full` removes and adds border around note content appropriately and immediately
```

**Review by mokagio — APPROVED** [review]

**Comment by pachlava** [comment]
@mokagio thank you for nitpicks! I'll add `.` to all steps in the next iteration of checklist update, which will happen today.

Do you know why the `Verify App Store Target Builds` job never gets done, and the only way to proceed with the merge for me here is force it?

**Comment by mokagio** [comment]
@pachlava You did well to force the merge, this PR only changed doc files anyways.

I'm pretty sure the reason was that the PR was created from a commit that had the step named "Build App Store Target" but [now it's name is "Verify..."](https://github.com/Automattic/simplenote-macos/blob/b74cb8d43efa8ffd68347c7ca2e8f1e8a996f255/.circleci/config.yml#L43-L50) and that's what GitHub uses to identify the step (as opposed to some UUID that maybe the CI provider should send, so that this kind of issue with renames doesn't occur).

Another solution could have been merge `develop` onto this, or (even better imho) rebase this on top of it.

**Pull request merged**